### PR TITLE
Name service client

### PR DESCRIPTION
**Pull request by tiago18c**
| Status  | Type  | ⚠️ Core Change | Issue |
| :---: | :---: | :---: | :--: |
| Ready | Feature | No | - |

## Problem

No way to query name records.

## Solution

Created a NameServiceClient that abstracts querying for name records.
This client allows to Query for:
- .sol name to address
- reverse address to owned .sol names
- twitter handle to address
- address to twitter handle
- all names owned by address
- mint address to token info
- token ticker to mint address

### CODE REVIEW

File: src/Solnet.Programs/Clients/NameServiceClient.cs [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+        /// <summary>
+        /// Gets all records owned by a given account.
+        /// </summary>
+        /// <param name="address">The owner address.</param>
+        /// <returns>A list containing all records.</returns>
+        public async Task<List<RecordBase>> GetAllNamesByOwnerAsync(string address)
+        {
+            var res = await RpcClient.GetProgramAccountsAsync(ProgramID, Rpc.Types.Commitment.Confirmed, null,
+                new List<MemCmp>() { new MemCmp() { Bytes = address, Offset = 32 }/*, new MemCmp() { Bytes = ReverseSolTLD, Offset = 64 }*/ });
```
Comment (murlokito) [inline src/Solnet.Programs/Clients/NameServiceClient.cs:67]: Commented code

File: src/Solnet.Programs/Clients/NameServiceClient.cs [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+        /// <summary>
+        /// Get name records owned by a given address.
+        /// </summary>
+        /// <param name="address">The owner address.</param>
+        /// <returns>A collection of name records owned by the address.</returns>
+        public async Task<List<ReverseNameRecord>> GetNamesFromAddressAsync(string address)
+        {
+
+            var res = await RpcClient.GetProgramAccountsAsync(ProgramID, Rpc.Types.Commitment.Confirmed, null,
+                new List<MemCmp>() { new MemCmp() { Bytes = SolTLD, Offset = 0 }, new MemCmp() { Bytes = address, Offset = 32 }/*, new MemCmp() { Bytes = ReverseSolTLD, Offset = 64 }*/ });
```
Comment (murlokito) [inline src/Solnet.Programs/Clients/NameServiceClient.cs:281]: More commented code

File: src/Solnet.Programs/Models/TokenData.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Solnet.Programs.Utilities;
+
+namespace Solnet.Programs.Models
+{
+    public class TokenData
```
Comment (murlokito) [inline src/Solnet.Programs/Models/TokenData.cs:11]: No comments here?

**Review by murlokito — APPROVED** [review]
Other than the comments I don't see anything that needs to be changed. LGTM

File: src/Solnet.Programs/Models/TokenData.cs (hunk 3 above)
Comment (tiago18c) [reply src/Solnet.Programs/Models/TokenData.cs:11]: Fixed and moved file to folder with remaining NameServiceClient related models.

**Pull request merged**